app/routes/app._index.jsx

```diff
@@ -132,9 +123,12 @@ export default function Index() {
       </ui-title-bar>
       <Layout>
         <Layout.Section>
-          <Card padding={"0"}>
-            {emptyMarkup}
-            {qrCodesMarkup}
+          <Card padding="0">
+            {qrCodes.length === 0 ? (
+              <EmptyQRCodeState onAction={() => navigate("qrcodes/new")} />
+            ) : (
+              <QRTable qrCodes={qrCodes} />
+            )}
```

Comment:
This is the result of splitting out the components - you can how easily tell from the JSX which component will be used, and why.